sklearn/externals/joblib/numpy_pickle.py

```diff
@@ -1,600 +1,3 @@
-    obj = None
-    try:
-        obj = unpickler.load()
-        if unpickler.compat_mode:
-            warnings.warn("The file '%s' has been generated with a "
-                          "joblib version less than 0.10. "
-                          "Please regenerate this pickle file."
-                          % filename,
-                          DeprecationWarning, stacklevel=3)
-    except UnicodeDecodeError as exc:
-        # More user-friendly error message
-        if PY3_OR_LATER:
-            new_exc = ValueError(
-                'You may be trying to read with '
-                'python 3 a joblib pickle generated with python 2. '
-                'This feature is not supported by joblib.')
-            new_exc.__cause__ = exc
-            raise new_exc
-        # Reraise exception with Python 2
-        raise
-
-    return obj
-
-
-def load(filename, mmap_mode=None):
-    """Reconstruct a Python object from a file persisted with joblib.dump.
-
-    Read more in the :ref:`User Guide <persistence>`.
-
-    Parameters
-    -----------
-    filename: str, pathlib.Path, or file object.
-        The file object or path of the file from which to load the object
-    mmap_mode: {None, 'r+', 'r', 'w+', 'c'}, optional
-        If not None, the arrays are memory-mapped from the disk. This
-        mode has no effect for compressed files. Note that in this
-        case the reconstructed object might no longer match exactly
-        the originally pickled object.
-
-    Returns
-    -------
-    result: any Python object
-        The object stored in the file.
-
-    See Also
-    --------
-    joblib.dump : function to save an object
-
-    Notes
-    -----
-
-    This function can load numpy array files saved separately during the
-    dump. If the mmap_mode argument is given, it is passed to np.load and
-    arrays are loaded as memmaps. As a consequence, the reconstructed
-    object might not match the original pickled object. Note that if the
-    file was saved with compression, the arrays cannot be memmapped.
-    """
-    if Path is not None and isinstance(filename, Path):
-        filename = str(filename)
-
-    if hasattr(filename, "read"):
-        fobj = filename
-        filename = getattr(fobj, 'name', '')
-        with _read_fileobject(fobj, filename, mmap_mode) as fobj:
-            obj = _unpickle(fobj)
-    else:
-        with open(filename, 'rb') as f:
-            with _read_fileobject(f, filename, mmap_mode) as fobj:
-                if isinstance(fobj, _basestring):
-                    # if the returned file object is a string, this means we
-                    # try to load a pickle file generated with an version of
-                    # Joblib so we load it with joblib compatibility function.
-                    return load_compatibility(fobj)
-
-                obj = _unpickle(fobj, filename, mmap_mode)
-
-    return obj
+from joblib.numpy_pickle import *
```

Comment:
This was added to allow loading pickles generated with older versions of scikit-learn, used e.g. in `fetch_20newsgroups_vectorized`.